File: samples/cloud/src/main/java/com/example/webflux/WebfluxApplication.java
package com.example.webflux;

import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;

@SpringBootApplication
@RestController
public class WebfluxApplication {

	public static void main(String[] args) {
		SpringApplication.run(WebfluxApplication.class, args);
	}

	@GetMapping("/")
	public String greet() {
		return "hi!";
	}

}

File: spring-graalvm-native-samples/webflux-kotlin/src/main/kotlin/com/example/webflux/WebfluxApplication.kt
package com.example.webflux

import kotlinx.coroutines.delay
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.runApplication
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RestController

@SpringBootApplication
class WebfluxApplication

fun main(args: Array<String>) {
	runApplication<WebfluxApplication>(*args)
}

data class Bar(val message: String)

@RestController
class WebfluxController {

	@GetMapping("/")
	suspend fun greet(): Bar {
		delay(10)
		return Bar("hi!")
	}
}
